src/lru.js

```diff
@@ -91,16 +115,12 @@ class LRU {
 		return this.items.keys();
 	}
 
-	set (key, value, bypass = false, resetTtl = this.resetTtl) {
-		let item;
-
-		if (bypass || this.items.has(key)) {
-			item = this.items.get(key);
+	set (key, value) {
+		// Reset existing item
+		if (this.items.has(key)) {
+			const item = this.items.get(key);
 			item.value = value;
-
-			if (resetTtl) {
-				item.expiry = this.ttl > 0 ? Date.now() + this.ttl : this.ttl;
-			}
+			item.expiry = this.ttl > 0 ? Date.now() + this.ttl : this.ttl;
```

Comment:
it is safe to always reseset ttl now, because set is never invoked from get anymore